syntax = "proto3";

package distributed;

service GrpcService {
    rpc adjustResCPU (ResRequest) returns (ResResponse) {} // adjust CPU resources
    rpc adjustResMEM (ResRequest) returns (ResResponse) {} // adjust MEM resources
}

message ResRequest {
    //map, format of value is "cpu_usage,replicas_number"
    repeated string uids = 1;   
    double value = 2;   
};

message ResRequest_2 {
    //map, format of value is "cpu_usage,replicas_number"
    repeated string uids = 3;   
    repeated double values = 4;   
};


message ResResponse {
    string result = 1;
};